src/elements/content-sharing/ContentSharing.js

```diff
@@ -110,30 +145,122 @@ function ContentSharing({ apiHost, displayInModal, itemID, itemType, language, t
         }
     }, [api, getError, item, itemID, itemType, sharedLink, currentUserID]);
 
-    if (errorMessage) {
-        return <ErrorMask errorHeader={<FormattedMessage {...errorMessage} />} />;
+    // Close a notification
+    const handleNotificationClose = React.useCallback(
+        (id: number) => {
+            const updatedNotifications = { ...notifications };
+            delete updatedNotifications[id];
+            setNotifications(updatedNotifications);
+        },
+        [notifications],
+    );
+
+    // Create a notification
+    const createNotification = React.useCallback(
+        (notificationType: NotificationType, message: StringMap) => {
```

Comment:
can we use `import type { MessageDescriptor } from 'react-intl';` for `message`?